src/main/java/utils/SchemaMetaData.java

```diff
@@ -2,37 +2,72 @@
 
 import java.sql.*;
 import java.util.ArrayList;
+import java.util.Collection;
 import java.util.List;
+import java.util.Optional;
 
 public class SchemaMetaData implements AutoCloseable {
     private final Connection connection;
 
     public class ColumnInfo {
         public String columnName;
         public String tableName;
-        public String referencedColumnName;
-        public String referencedTableName;
-        public ColumnInfo(String columnName,
-                          String tableName,
-                          String referencedColumnName,
-                          String referencedTableName) {
+
+        public ColumnInfo(String columnName, String tableName) {
             this.columnName = columnName;
             this.tableName = tableName;
-            this.referencedColumnName = referencedColumnName;
-            this.referencedTableName = referencedTableName;
         }
 
         @Override
         public String toString() {
             return "ColumnInfo{" +
                     "columnName='" + columnName + '\'' +
                     ", tableName='" + tableName + '\'' +
-                    ", referencedColumnName='" + referencedColumnName + '\'' +
-                    ", referencedTableName='" + referencedTableName + '\'' +
                     '}';
         }
     }
 
+    public class FKColumnInfo {
```

Comment:
FKColumnInfo sugeruje, jakby ta klasa dziedziczyła po ColumnInfo, co jest mylące. Może zamiast tego ForeignKeyInfo?

Reply:
Nie spojrzałem na to w ten sposób, ale rzeczywiście teraz tak to wygląda.